compiler/damlc/tests/daml-test-files/InterfaceDoubleSpend.daml

```diff
@@ -0,0 +1,56 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- @SINCE-LF-FEATURE DAML_INTERFACE
+
+module InterfaceDoubleSpend where
+
+-- | An interface comment.
+interface Token where
+  getIssuer : Party
+  getOwner : Party
+  getAmount : Int
+
+  transferImpl : Party -> Update (ContractId Token)
+
+  ensure (getAmount this >= 0)
+
+  choice Transfer : ContractId Token
+    with
+      newOwner : Party
+    controller getIssuer this
+    do
+      transferImpl this newOwner
+
+template Asset
+  with
+    issuer : Party
+    owner : Party
+    amount : Int
+  where
+    signatory issuer
+    observer owner
+    implements Token where
+      let getOwner = owner
+      let getIssuer = issuer
+      let getAmount = amount
+      let transferImpl = \newOwner -> do
+            cid <- create this with owner = newOwner
+            pure (toInterfaceContractId @Token cid)
+
+main = scenario do
```

Comment:
Is there a reason why you’re testing this at this level rather than at the Engine level?

Reply:
1. This test should go in `EvaluationOrderTest`, i think it is better to put it there when we implement all order test for FetchInterface.

Reply:
I will add a todo here. If you think we should add it right now in LF scope, I will do it.